src/components/data-exploration/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -61,18 +61,31 @@ const HeatmapMetadataTrackSettings = (props) => {
     setTrackData(getUpdatedTrackData());
   }, [cellSets.hierarchy]);
 
+  const getEnabledTracks = () => trackData.filter((entry) => entry.selected).map((o) => o.key);
+
+  useEffect(() => {
+    if (!_.isEqual(getEnabledTracks(), selectedTracks)) {
+      const newTracks = trackData.map((entry) => {
+        if (!selectedTracks.includes(entry.key)) {
+          return { ...entry, selected: false };
+        }
+        return { ...entry, selected: true };
+      });
+      setTrackData(newTracks);
+    }
+  }, [selectedTracks]);
```

Comment:
Does this handle some new use case?

Reply:
Yes, when we change the data we want the group by to be changed and metadata tracks too